src/windows_registry.rs

```diff
@@ -110,19 +115,96 @@ pub fn find_tool(target: &str, tool: &str) -> Option<Tool> {
     // environment variables like `LIB`, `INCLUDE`, and `PATH` to ensure that
     // the tool is actually usable.
 
-    return find_msvc_latest(tool, target, "15.0")
-        .or_else(|| find_msvc_latest(tool, target, "14.0"))
+    return find_msvc_17(tool, target)
+        .or_else(|| find_msvc_14_or_15(tool, target, "15.0"))
+        .or_else(|| find_msvc_14_or_15(tool, target, "14.0"))
         .or_else(|| find_msvc_12(tool, target))
         .or_else(|| find_msvc_11(tool, target));
 
-    // For MSVC 14 or newer we need to find the Universal CRT as well as either
+    // In MSVC 17 MS once again changed the scheme for locating the tooling.
+    // Now we must go through some COM interfaces, which is super fun for Rust.
+    fn find_msvc_17(tool: &str, target: &str) -> Option<Tool> {
```

Comment:
Note that while it is called VS 2017, the version number is actually 15, not 17.